ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -536,8 +536,10 @@ public Wei getUpfrontGasCost() {
    * @return the up-front cost for the gas the transaction can use.
    */
   public Wei getUpfrontGasCost(final Wei gasPrice) {
-    return Wei.of(getGasLimit())
-        .multiply(Wei.of(getMaxFeePerGas().orElse(gasPrice).getAsBigInteger()));
+    if (gasPrice == null || gasPrice.isZero()) {
+      return Wei.ZERO;
+    }
```

Comment:
I'm trying to find where this could be null from? I'm hoping this isn't test code concerns leaking into production code.

Reply:
I mostly revert this part because it should not change with my fix. but I think that if we have to remove this check it will be preferable to remove it in another PR non EIP1559 in order to identify it more easily if a regression that we would not have seen appears

Reply:
it could logically be null though right?  gasPrice should be absent from a 1559 transaction

Reply:
Yes I think it can maybe possible, I just had a usecase when the gaslimit is null